backend/src/main/java/storage/repository/GenericRepository.java

```diff
@@ -0,0 +1,240 @@
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /***
+     * Gets specific instance of model
+     * @param id
+     * @return
+     */
+    public T getObject(int id) {
+        String query = "SELECT * FROM "+ type.getSimpleName() +" WHERE id="+id;
+        System.out.println("Running query: "+query);
+        ResultSet rs =  this.dbInterface.executeQuerry(query);
+        try {
+            return this.getMatchJSON(rs).iterator().next(); //converts resulting set to model instance
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /***
+     * Creates an given instance om model
+     * @param object
+     * @return
+     */
+    public T postObject(T object) {
+        JsonObject jsonObject = this.gson.toJsonTree(object).getAsJsonObject();
+        StringBuilder query = new StringBuilder("INSERT INTO "+ type.getSimpleName() +" (");
+
+        for (String key : jsonObject.keySet()){
+            if(Objects.equals(key, "id"))continue; //skip ID
+            String last = query.substring(query.length() - 1);
+            if(!last.equals("(")){ //first
+                query.append(", ");
+            }
+            query.append(key);
+        }
+        query.append(")");
+
+        for(int i = 0; i < jsonObject.keySet().size(); i++){
+            query.append(i==0?" VALUES (":i==1?"?":",?"); //skip ID
+        }
+        query.append(")");
+
+        try {
+            PreparedStatement ps = getPreparedStatementFromJSON(jsonObject, query); //maps model data to the query
+            ps.executeUpdate();
+            ResultSet rs = ps.getGeneratedKeys();
+            rs.next();
+            int id = rs.getInt(1);
+            return this.getObject(id);
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    /***
+     * Updates a model already in the system, currently updates ALL fields no matter how many are set (probably)
+     * @param object
+     * @return
+     */
+    public T updateObject(T object) {
+        JsonObject jsonObject = this.gson.toJsonTree(object).getAsJsonObject();
+        StringBuilder query = new StringBuilder("UPDATE "+ type.getSimpleName()+" SET");
+
+        for (String key : jsonObject.keySet()){
+            if(Objects.equals(key, "id"))continue; //skip id
+            String last = query.substring(query.length() - 1);
+            query.append(
+                last.equals("T")?
+                    " "+key :
+                    " = ?, "+key);
+        }
+        query.append(" = ? ");
+        query.append( " WHERE id="+object.getId());
+
+        try {
+            PreparedStatement ps = getPreparedStatementFromJSON(jsonObject, query); //maps model data to the query
+            ps.executeUpdate();
+            ResultSet rs = ps.getGeneratedKeys();
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "ResultSet" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)